src/eterna/mode/PoseEdit/PoseEditMode.ts

```diff
@@ -2152,25 +2159,58 @@ export default class PoseEditMode extends GameMode {
                 libraryNT: this._poses[0].librarySelections ?? []
             }, solToSubmit);
         } else {
-            const NOT_SATISFIED_PROMPT = 'Puzzle constraints are not satisfied.\n\n'
-                + 'You can still submit the sequence, but please note that there is a risk of not getting '
-                + 'synthesized properly';
-
-            if (!this.checkConstraints()) {
-                // If we pass constraints when taking into account soft constraints, just prompt
-                if (this.checkConstraints(this._puzzle.isSoftConstraint || Eterna.DEV_MODE)) {
-                    this.showConfirmDialog(NOT_SATISFIED_PROMPT).closed
-                        .then((confirmed) => {
-                            if (confirmed) {
-                                this.promptForExperimentalPuzzleSubmission();
+            const pipeline = [
```

Comment:
I think this is a reasonable approach for now, but this pipeline here is an implicit state machine.

Reply:
It is, and I debated creating a utility class or something, but seemed like more overhead than necessary for the one small case this shows

Reply:
Agreed. Just want it noted since I keep pushing for adopting a state machine approach eventually to improve some business logic flexibility and maintainability.